SCRUM-201: Initial setup of TypeORM.

aas1984 opened the pull request:
Created module for database and many of the major entities in the system. Set up relations for all of them and tested manually on both Postman and pgAdmin 4. Many of the create() functions in the service classes use hardcoded values. I'm aware these need to change later, but they work for now.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6)
<!-- Reviewable:end -->

aas1984 committed:
Added PostgreSQL and TypeORM to Node

A bunch of "dev": true got set to false, I'm not enough of a node guy to know if this is an issue or a way to avoid it.

aas1984 committed:
Added PostgreSQL and TypeORM to Node

A bunch of "dev": true got set to false, I'm not enough of a node guy to know if this is an issue or a way to avoid it.

aas1984 committed:
Set up initial modules

aas1984 committed:
Set up initial modules

aas1984 committed:
Added initial User module.

aas1984 committed:
Added initial User module.

aas1984 committed:
Fixed issue with \inventory POST

Will need to be added back later when CreateInventoryDto is implemented.

aas1984 committed:
Fixed issue with \inventory POST

Will need to be added back later when CreateInventoryDto is implemented.

aas1984 committed:
Added character limits to inventory table

aas1984 committed:
Added character limits to inventory table

aas1984 committed:
Removed unnecessary comments

aas1984 committed:
Removed unnecessary comments

aas1984 committed:
Removed unnecessary import

aas1984 committed:
Removed unnecessary import

aas1984 committed:
Added initial Organization module

aas1984 committed:
Added initial Organization module

aas1984 committed:
Added initial invite module

aas1984 committed:
Added initial invite module

aas1984 committed:
Added initial Request module

aas1984 committed:
Added initial Request module

aas1984 committed:
Set up initial Review module

aas1984 committed:
Set up initial Review module

aas1984 committed:
Added initial Conversation module

aas1984 committed:
Added initial Conversation module

aas1984 committed:
Setup initial Message module

aas1984 committed:
Setup initial Message module

aas1984 committed:
Setup initial Bookmark module

aas1984 committed:
Setup initial Bookmark module

aas1984 committed:
Added initial Listing module

aas1984 committed:
Added initial Listing module

aas1984 committed:
Setup initial Order module

aas1984 committed:
Setup initial Order module

aas1984 committed:
Setup initial InventoryItem module

aas1984 committed:
Setup initial InventoryItem module

aas1984 committed:
Added relation between User and Organization

aas1984 committed:
Added relation between User and Organization

aas1984 committed:
Added relation between user and itself for referrals

aas1984 committed:
Added relation between user and itself for referrals

aas1984 committed:
Added relation between organization and user

aas1984 committed:
Added relation between organization and user

aas1984 committed:
Added relation between inventory and organization

aas1984 committed:
Added relation between inventory and organization

aas1984 committed:
Removed unique username to make testing easier

aas1984 committed:
Removed unique username to make testing easier

aas1984 committed:
style: renamed variables and changed some types

EIN and ZIP were changed to string
id variables were renamed for clarity
timestamps were changed to date
added semi-colons where they were missing

aas1984 committed:
style: renamed variables and changed some types

EIN and ZIP were changed to string
id variables were renamed for clarity
timestamps were changed to date
added semi-colons where they were missing

aas1984 committed:
Added relations for Order

aas1984 committed:
Added relations for Order

aas1984 committed:
Removed parentInventory from inventory

aas1984 committed:
Removed parentInventory from inventory

aas1984 committed:
Added relations to Invite

aas1984 committed:
Added relations to Invite

aas1984 committed:
Added relations for Review

aas1984 committed:
Added relations for Review

aas1984 committed:
Added relations for Request

aas1984 committed:
Added relations for Request

aas1984 committed:
Added relations for Message

aas1984 committed:
Added relations for Message

aas1984 committed:
Added relations to Listing

aas1984 committed:
Added relations to Listing

aas1984 committed:
Added relation to InventoryItem

aas1984 committed:
Added relation to InventoryItem

aas1984 committed:
Added relation for Conversation

aas1984 committed:
Added relation for Conversation

aas1984 committed:
Improved error messages when testing

aas1984 committed:
Improved error messages when testing

aas1984 committed:
Removed unnecessary imports

aas1984 committed:
Removed unnecessary imports

notjoncruz committed:
feat: adding husky to run lint and tests

aas1984 committed:
Making the linter happy

aas1984 committed:
Making the linter happy

aas1984 committed:
bugfix: fixed test suite

Changed every absolute import to a relative one, and added mocks for every repository in the test files. Ideally in the future these mocks are more centralized so less code is duplicated.

aas1984 committed:
bugfix: fixed test suite

Changed every absolute import to a relative one, and added mocks for every repository in the test files. Ideally in the future these mocks are more centralized so less code is duplicated.

Jsd5582 commented on src/inventory-item/inventory-item.entity.ts:
```diff
@@ -0,0 +1,35 @@
+    @Column({ type: "varchar", length: 4000 })
+    attributes: string;
+
+    @OneToMany(() => Listing, listing => listing.inventoryItem)
```
Why does inventory item have a 1 : many, thought it was 1 : 1

aas1984 replied:
Because an inventory item can have multiple listings from it. 
<!-- Reviewable comment -OHdDLgq-BEr6Kup5aj2:b-c59rpd -->
<!-- Sent from Reviewable.io -->

Jsd5582 commented on src/inventory/inventory.entity.ts:
```diff
@@ -0,0 +1,26 @@
+    @Column({ type: "varchar", length: 4000 })
+    description: string;
+
+    @Column({ type: "varchar", length: 100 }) 
```
Shouldn't this be split into the address line 1, address line 2, zipcode, State, etc

aas1984 replied:
Good call, I made the change.

Jsd5582 commented on src/organization/organization.entity.ts:
```diff
@@ -0,0 +1,69 @@
+    rating: number;
+
+    @Column({ type: 'varchar', length: 50 })
+    address1: string;
```
Maybe name it addressLine1 & addressLine2 just so it doesn't seem like its saving 2 separate addresses

aas1984 replied:
Good call, made the change.

Jsd5582 commented on src/order/order.entity.ts:
```diff
@@ -0,0 +1,45 @@
+    listing: Listing;
+
+    @JoinColumn()
+    @ManyToOne(() => Organization, org => org.orders)
```
wouldn't the owner of an order be the User who pulls it from the "pool"? We might have already discussed this but I'm not completely sure what we decided

aas1984 replied:
You're right, I accidentally copied it over wrong from the create.sql. Fixed!

Jsd5582 commented on src/user/user.entity.ts:
```diff
@@ -0,0 +1,88 @@
+    @Column({ default: false })
+    inactive: boolean;
+
+    @Column({ type: "varchar", default: '0000000000', length: 10 })
```
I don't know if there's a way to but can we make this auto generated?

notjoncruz replied:
yeah, lets look into this.  I don't think we should specify a default
<!-- Reviewable comment -OHcxy9Q0a0AYahZSdlS:b-7rvjqa -->
<!-- Sent from Reviewable.io -->

aas1984 replied:
I've looked into it, I couldn't find any real way to do it on postgres or with typeorm. 
<!-- Reviewable comment -OHdAQk2AFdHHSDqzzSX:b8h3c52 -->
<!-- Sent from Reviewable.io -->

aas1984 committed:
fix: updated PR with comments from reviews

Changed address1 and address2 in organization to addressLine1 and addressLine2.
Updated location to full address data in inventory.
Moved message folder into conversation and inventory-item folder into inventory, then updated all imports.

aas1984 committed:
fix: updated PR with comments from reviews

Changed address1 and address2 in organization to addressLine1 and addressLine2.
Updated location to full address data in inventory.
Moved message folder into conversation and inventory-item folder into inventory, then updated all imports.

Jsd5582 commented on src/review/review.entity.ts:
```diff
@@ -0,0 +1,32 @@
+    order: Order;
+
+    @Column({ type: "varchar", length: 4000 }) 
+    description: string;
```
Maybe change description to content

notjoncruz replied:
I agree
<!-- Reviewable comment -OHcxRFf9HBfX8XY6APC:b-cstnkg -->
<!-- Sent from Reviewable.io -->

aas1984 replied:
Done.
<!-- Reviewable comment -OHd9kkX6dkcDAXxi1X0:b-896fix -->
<!-- Sent from Reviewable.io -->

Commit:
Merge pull request #7 from The-Mobility-Independence-Foundation/add-husky

feat: adding husky to run lint and tests

notjoncruz commented on package.json:
```diff
@@ -25,8 +25,12 @@
     "@nestjs/platform-express": "^10.0.0",
     "@nestjs/swagger": "^8.1.1",
+    "@nestjs/typeorm": "^10.0.2",
+    "pg": "^8.13.1",
```
why do we need `pg`, `postgres` and `rxjs`?

aas1984 replied:
I did not add rxjs, I do not know why it is here.
<!-- Reviewable comment -OHdCuy18u4CAO-4V04W:bsd4iun -->
<!-- Sent from Reviewable.io -->

notjoncruz reviewed (changes requested):
most comments are on the entities. for each module, can you create a folder `entities` and place those `.entity.ty` file in there? we will do the same for dtos

> Reviewed 53 of 79 files at r1, 26 of 26 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-:-OHcyvYI8fGrdDuKIfAv:bmhvda2)* status: all files reviewed, 46 unresolved discussions (waiting on @aas1984 and @Jsd5582)

___
*[`src/bookmark/bookmark.controller.spec.ts` line 22 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcPdUNCvpwQmqwPS4y:-OHcPdUNCvpwQmqwPS4z:bmnadtq) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.controller.spec.ts#L22)):*
> ```XML
>       providers: [
>         BookmarkService, 
>         {
> ```

can you explain what's happening here?
___
*[`src/bookmark/bookmark.controller.spec.ts` line 24 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcP62x8a8LoJSE9YnV:-OHcP62x8a8LoJSE9YnW:b59rkr9) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.controller.spec.ts#L24)):*
> ```XML
>         {
>           provide: getRepositoryToken(Bookmark), 
>           useClass: mockRepository
> ```

can we define the `jest.fn()` here rather than defining it as a constant above? 
___
*[`src/bookmark/bookmark.entity.ts` line 9 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcPk3c53_iXs0zPwFz:-OHcPk3c53_iXs0zPwG-:b9ieis) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.entity.ts#L9)):*
> ```XML
>     id: number;
> 
>     @Column({ nullable: false }) // TODO: foreign key on user
> ```

do we have to explicitly define columns that are not nullable?
___
*[`src/bookmark/bookmark.service.spec.ts` line 22 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcS_S7BzQumlknGYcJ:-OHcS_S7BzQumlknGYcK:b-p8tsms) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.service.spec.ts#L22)):*
> ```XML
>         {
>           provide: getRepositoryToken(Bookmark), 
>           useClass: mockRepository
> ```

same here and for all places, I think we should inline define the `jest.fn()`
___
*[`src/inventory/inventory.entity.ts` line 14 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcWRCS9Exautx9DDXp:-OHcWRCS9Exautx9DDXq:b45wuwx) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/inventory/inventory.entity.ts#L14)):*
> ```XML
>     organization: Organization;
> 
>     @Column({ type: "varchar", length: 40 })
> ```

again, any reasoning for all these defined lengths?
___
*[`src/inventory/inventory.entity.ts` line 20 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcWftqB6Rwn1q1oaHP:-OHcWftr-l9zM30IldPc:b-k7fxp3) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/inventory/inventory.entity.ts#L20)):*
> ```XML
>     description: string;
> 
>     @Column({ type: 'varchar', length: 50 })
> ```

not too sure if 50 characters is enough for address line 1
___
*[`src/conversation/conversation.module.ts` line 9 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcUO376u5tlnyCCUGJ:-OHcUO376u5tlnyCCUGK:b-ywdbdm) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/conversation.module.ts#L9)):*
> ```XML
> 
> @Module({
>   imports: [TypeOrmModule.forFeature([Conversation, Listing])],
> ```

shouldn't the message module be in here as well?
___
*[`src/database/database.module.ts` line 19 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcVfJG6mp3RQsgrUHs:-OHcVfJG6mp3RQsgrUHt:bvz65xi) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/database/database.module.ts#L19)):*
> ```XML
> // When we set up the actual DB, we need to change this to read values from a file. 
> // That way, we aren't leaking db credentials on our public git repo.
> @Module({
> ```

shouldn't all this logic be stored in `app.module.ts`?
___
*[`src/database/database.module.ts` line 20 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcVtQ94GsItrPOkMW1:-OHcVZq23u-hi6fALTNT:b-5eqv6f) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/database/database.module.ts#L20)):*
> ```XML
> // That way, we aren't leaking db credentials on our public git repo.
> @Module({
>     imports: [TypeOrmModule.forRoot({
> ```

I see your comment above. this will have to be updated to something like this

```
TypeOrmModule.forRootAsync({
      imports: [ConfigModule],
      useFactory: (config: ConfigService) => ({
            host: config.get('DATABASE_HOST'),
            ...
      })
      inject: [ConfigService],
}),
```
___
*[`src/inventory/interfaces/inventory.interface.ts` line 1 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcWtdA0zcCmYc5wcsU:-OHcWtdA0zcCmYc5wcsV:b-ef5xw4) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/inventory/interfaces/inventory.interface.ts#L1)):*
> ```XML
> // I don't think this is currently needed but I'm leaving it here for now, can delete later.
> ```

if its not needed, I would just delete this so that there's no confusion in the future
___
*[`src/bookmark/bookmark.controller.ts` line 8 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcOXP-3A0rqb9mEv0f:-OHcOXP-3A0rqb9mEv0g:bw9pic8) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.controller.ts#L8)):*
> ```XML
> export class BookmarkController {
> 
>     constructor(private bookmarkService: BookmarkService) {}
> ```

nit: can you define these as `private readonly`
___
*[`src/user/user.entity.ts` line 17 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcxgQPDCEBn0wCNVFF:-OHcxgQPDCEBn0wCNVFG:b-i7fzn7) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L17)):*
> ```XML
> 
> @Entity()
> export class User {
> ```

how will password and email look if the user is using a different form of authentication?
___
*[`src/user/user.entity.ts` line 17 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcyoLu1khFodmQXEWB:-OHcyoLv70iCg-04fmgq:b3m3omi) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L17)):*
> ```XML
> 
> @Entity()
> export class User {
> ```

I think conversations is missing
___
*[`src/user/user.entity.ts` line 44 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcxnak0AYh0gY9ZHtv:-OHcxnal7XOIiuS-K_N3:b-2kjrqk) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L44)):*
> ```XML
> 
>     @Column({ type: "enum", enum: UserRole, default: UserRole.USER })
>     accType: UserRole;
> ```

change this to `type`
___
*[`src/user/user.entity.ts` line 50 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcxqhI9P9ziPyhU-Ut:-OHcxqhI9P9ziPyhU-Uu:b-hyznqy) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L50)):*
> ```XML
> 
>     @Column({ default: false })
>     inactive: boolean;
> ```

do we need a field to determine if the user is inactive?
___
*[`src/user/user.entity.ts` line 66 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcyaI405O8ZFl1-ZuY:-OHcyBdE9p8GagRz17h-:bab85fu) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L66)):*
> ```XML
> 
>     @OneToMany(() => User, user => user.referredBy)
>     referrals: User[];
> ```

what will be the difference between `referrals` and `sentInvites`? 
___
*[`src/user/user.entity.ts` line 84 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcyYEo59laBC2XGysy:-OHcyYEo59laBC2XGysz:btmrx0v) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L84)):*
> ```XML
> 
>     @OneToMany(() => Request, request => request.approver)
>     approvedRequests: Request[];
> ```

why is request an array?
___
*[`src/user/user.entity.ts` line 87 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcykHL4EqcvByf3q8k:-OHcykHL4EqcvByf3q8l:bghjmxr) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L87)):*
> ```XML
> 
>     @OneToMany(() => Message, message => message.sender)
>     sentMessages: Message[];
> ```

is this necessary? 
___
*[`src/user/user.entity.ts` line 90 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcygO31LIYvy4JdswZ:-OHcygO31LIYvy4Jdsw_:b7bqt6y) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L90)):*
> ```XML
> 
>     @OneToMany(() => Listing, listing => listing.owner)
>     listings: Listing[];
> ```

does a user have listings, or does an organization?
___
*[`src/invite/invite.module.ts` line 10 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcvrI931HXjBaj10k8:-OHcvrIABUm7UFCRg9A5:bspkdpg) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.module.ts#L10)):*
> ```XML
> 
> @Module({
>   imports: [TypeOrmModule.forFeature([Invite, User, Organization])],
> ```

why would the invite module need user and organization?
___
*[`src/conversation/message/message.module.ts` line 10 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcVBQH3WuHUs-l50KX:-OHcVBQH3WuHUs-l50KY:bqwp7gf) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.module.ts#L10)):*
> ```XML
> 
> @Module({
>   imports: [TypeOrmModule.forFeature([Message, User, Conversation])],
> ```

why would the message module need the user and conversation module?
___
*[`src/organization/organization.entity.ts` line 20 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwUUl4OpIB0QceaNP:-OHcwUUmCU9O5YuNb7pY:bcp5kyo) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L20)):*
> ```XML
>     owner: User;
> 
>     // returns a 500 server error unless we catch the duplicate name error ourselves. 
> ```

we should be fine if organizations have the same name. I think we will want to find something else to differentiate them, like tax id
___
*[`src/organization/organization.entity.ts` line 26 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwbfJ8j4f3wndViLR:-OHcwbfJ8j4f3wndViLS:b1hkmuq) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L26)):*
> ```XML
> 
>     @Column({ type: "timestamp", default: () => "CURRENT_TIMESTAMP" })
>     lastActivity: Date;
> ```

what does this refer to?
___
*[`src/organization/organization.entity.ts` line 32 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwgBs-nXW7OOe-c6s:-OHcwgBs-nXW7OOe-c6t:bbgsemh) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L32)):*
> ```XML
> 
>     @Column({ type: "varchar", length: 4000, default: "" })
>     services: string;
> ```

how would this look like?
___
*[`src/organization/organization.entity.ts` line 35 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwkuJ05gq_2CrljEZ:-OHcwkuJ05gq_2CrljE_:bwb5s6w) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L35)):*
> ```XML
> 
>     @Column({ type: 'decimal', default: 0.0 })
>     rating: number;
> ```

how will this be updated?
___
*[`src/organization/organization.entity.ts` line 59 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwpRD41w2vPMgB7sv:-OHcwpRD41w2vPMgB7sw:b-eyd37a) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L59)):*
> ```XML
> 
>     @Column({ type: "varchar", length: 50, array: true, nullable: true })
>     socials: string[];
> ```

should this be a json?

```
{
    website: "...",
    twitter: "...",
    ....=
}
```
___
*[`src/user/dto/create-user.dto.ts` line 1 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcysQsEhegCS8w3Vb8:-OHcysQt5QA_7DkqT2gd:bid1b96) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/dto/create-user.dto.ts#L1)):*
> ```XML
> export class CreateUserDto {
> ```

delete it for now
___
*[`src/conversation/message/message.service.ts` line 22 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcVNK68215i-67wYvT:-OHcVNK7BlZfuZ6mjg4o:bv146cb) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.service.ts#L22)):*
> ```XML
>     ) {}
> 
>     async create() {
> ```

will this be updated in the future?
___
*[`src/listing/listing.entity.ts` line 36 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcw0ytCVIxx1O8NdWR:-OHcw0ytCVIxx1O8NdWS:b2x8gr6) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/listing/listing.entity.ts#L36)):*
> ```XML
>     attributes: string;
> 
>     @Column({ default: 1 })
> ```

there was a default of 0 for inventory item, I think. any reason why this is different?
___
*[`src/listing/listing.entity.ts` line 39 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcw7RQ1-WDj2ZDaYfv:-OHcw7RQ1-WDj2ZDaYfw:b670uhm) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/listing/listing.entity.ts#L39)):*
> ```XML
>     quantity: number;
> 
>     // TODO: research to see if theres a better way to handle location
> ```

I think this is how we want to do this
___
*[`src/invite/invite.entity.ts` line 5 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcXUt8CM0ULvjtoCDK:-OHcXUt8CM0ULvjtoCDL:bz92ixb) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L5)):*
> ```XML
> import { Entity, Column, PrimaryGeneratedColumn, JoinColumn, ManyToOne } from 'typeorm';
> 
> export enum InviteType {
> ```

im assuming this is how we are going to differentiate from user invites to an org, and org invites to the site. im not too sure if this is how we want to do this. 
___
*[`src/invite/invite.entity.ts` line 18 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcY4kgDczlzJZmZ7At:-OHcY4kgDczlzJZmZ7Au:b-mdnvx) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L18)):*
> ```XML
>     @JoinColumn() 
>     @ManyToOne(() => User, user => user.sentInvites)
>     sender: User;
> ```

maybe inviter?
___
*[`src/invite/invite.entity.ts` line 25 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcY8M27bI-Sk7ODczg:-OHcY8M27bI-Sk7ODczh:bz0ktlu) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L25)):*
> ```XML
> 
>     @Column({ type: "varchar", length: 32 })
>     recieverEmail: string;
> ```

maybe inviteeEmail?
___
*[`src/invite/invite.entity.ts` line 31 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcYC0s1qRFMYXpv_uh:-OHcYC0s1qRFMYXpv_ui:b7dflbw) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L31)):*
> ```XML
> 
>     @Column({ type: "enum", enum: InviteType, default: InviteType.ORGANIZATION })
>     invType: InviteType;
> ```

I think `type` would be fine
___
*[`src/invite/invite.entity.ts` line 36 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcYGSDB__6_2eyxH32:-OHcYGSDB__6_2eyxH33:b-prq6bi) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L36)):*
> ```XML
>     sentOn: Date;
> 
>     @Column({ type: "timestamp", default: () => "CURRENT_TIMESTAMP + INTERVAL '14 day'" }) // TODO: make it set expiry date properly
> ```

this is cool
___
*[`package.json` line 29 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcNSe58PLgTcL8sldT:-OHcNSe58PLgTcL8sldU:b-u05jiw) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/package.json#L29)):*
> ```JSON
>     "@nestjs/swagger": "^8.1.1",
>     "@nestjs/typeorm": "^10.0.2",
>     "pg": "^8.13.1",
> ```

why do we need `pg` and `postgres`?
___
*[`src/request/request.entity.ts` line 11 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcx8fM3kP9_yM27WsT:-OHcx8fM3kP9_yM27WsU:budlx0i) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/request/request.entity.ts#L11)):*
> ```XML
> 
> @Entity()
> export class Request {
> ```

should organization name be on here too? I wonder if we should include organization details here to easily populate those fields to create an entry if the request is approved.
___
*[`src/conversation/message/message.entity.ts` line 13 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcV0DEBTi9lVeMu_2E:-OHcV0DEBTi9lVeMu_2F:b-ocam3o) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.entity.ts#L13)):*
> ```XML
>     @JoinColumn()
>     @ManyToOne(() => User, user => user.sentMessages)
>     sender: User;
> ```

I wonder if we should change this `author`
___
*[`src/conversation/message/message.entity.ts` line 19 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcUeA6-PGg_8F2MoWV:-OHcUeA6-PGg_8F2MoWW:byfwpn0) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.entity.ts#L19)):*
> ```XML
>     conversation: Conversation;
> 
>     @Column({ type: "varchar", length: 4000 })
> ```

reasoning for this defined length?
___
*[`src/conversation/message/message.entity.ts` line 22 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcUv_k6z0d6ym2_wJh:-OHcUv_k6z0d6ym2_wJi:b-q0xfsd) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.entity.ts#L22)):*
> ```XML
>     messageContent: string;
> 
>     @Column({ type: "timestamp", default: () => "CURRENT_TIMESTAMP" })
> ```

why is readStatus a date?


<!-- Sent from Reviewable.io -->

aas1984 reviewed (commented):
We could do that but all of the folders only have one entity file.

> *[Reviewable](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-:-OHdA19XCSosrXTNX2-C:btbnqwo)* status: all files reviewed, 47 unresolved discussions (waiting on @aas1984, @Jsd5582, and @notcruz)

___
*[`package.json` line 29 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcNSe58PLgTcL8sldT:-OHdCy1M8CJfH9vDYlrG:b-o5b33x) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/package.json#L29)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

why do we need `pg` and `postgres`?
</blockquote></details>

pg is needed for typeorm to be able to connect to our postgres database. I'm not sure why postgres is here, I didn't add it manually so it may have been included alongside pg.
___
*[`src/bookmark/bookmark.controller.ts` line 8 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcOXP-3A0rqb9mEv0f:-OHd1L-nEnT-s5wgJvdP:b-hp9g4b) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.controller.ts#L8)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

nit: can you define these as `private readonly`
</blockquote></details>

Yes, I just tested them and nothing broke so I should be able to update them all.
___
*[`src/bookmark/bookmark.controller.spec.ts` line 22 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcPdUNCvpwQmqwPS4y:-OHd1WfxE4yjDKWXQGl1:b-qh490i) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.controller.spec.ts#L22)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

can you explain what's happening here?
</blockquote></details>

Nest wants us to make a fresh testing module to ensure that the testing environments are consistent. The controller needs the service to have anything worth testing, so we include that directly, but for the rest we inject a mock. In theory, we could choose to not do this if we wanted, but then we would have integration tests instead of unit tests, and we would need to preserve some rows of data in the DB to stop the tests from breaking. 
___
*[`src/bookmark/bookmark.controller.spec.ts` line 24 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcP62x8a8LoJSE9YnV:-OHd1yMw8iyVA2eiM4fQ:b-1xn1lq) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.controller.spec.ts#L24)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

can we define the `jest.fn()` here rather than defining it as a constant above?
</blockquote></details>

We can do that but then we would have a lot of duplicated code. What we should do is make some test_index.ts file that defines a whole bunch of constants that we can use and reuse throughout all of the tests. As an example, we could make a jest.fn() for an OrganizationRepository that returns some specific data, and then use organizationMockRepository in all of the files where its needed. I did not set that up yet, because I was just trying to make the autogenerated test files run without error, but the boilerplate is there for us to build off of in the future.
___
*[`src/bookmark/bookmark.entity.ts` line 9 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcPk3c53_iXs0zPwFz:-OHd2Mbq64EPSHxl9s98:b-unja4n) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.entity.ts#L9)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

do we have to explicitly define columns that are not nullable?
</blockquote></details>

Nullable is false by default for all columns. If we wanted a column to be nullable, we would have to explicitly define it as such with both { nullable: true } and user: number | null; 
___
*[`src/bookmark/bookmark.service.spec.ts` line 22 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcS_S7BzQumlknGYcJ:-OHd3V4C7dZY6S2syea-:b-8bq9aj) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/bookmark/bookmark.service.spec.ts#L22)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

same here and for all places, I think we should inline define the `jest.fn()`
</blockquote></details>

I think having a test index file where we declare constants that are re-used is the ideal way to set this up. It reduces code duplication by a lot and it's the strategy I used when mocking out DB calls at my first co-op. 
___
*[`src/conversation/conversation.module.ts` line 9 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcUO376u5tlnyCCUGJ:-OHd3j2wAALXiMjtovqe:blzojlt) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/conversation.module.ts#L9)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

shouldn't the message module be in here as well?
</blockquote></details>

The code in the service module is currently just set up to allow me to insert data into the db and check to make sure everything is there correctly. As a result, it does not currently query the DB for messages at all. When it does in the future, Message will need to be added. The reason I didn't add it already is because the linter complains about unused imports and variables, and the Message module would currently be unused. 
___
*[`src/database/database.module.ts` line 19 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcVfJG6mp3RQsgrUHs:-OHd5_0Q0yOtD-UckOwI:b-q3tpa1) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/database/database.module.ts#L19)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

shouldn't all this logic be stored in `app.module.ts`?
</blockquote></details>

We could store all of this in app.module if we wanted, but the Nest.JS recipe you sent me for TypeORM explicitly said to set it up like this. I believe the main reason for this is to allow for multiple db connections easier, but we could move all of these imports into app.module if we wanted. I just think it would make things clunkier. 
___
*[`src/database/database.module.ts` line 20 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcVtQ94GsItrPOkMW1:-OHdD7Vv-ExKgw5o4ffg:b-19qcph) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/database/database.module.ts#L20)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

I see your comment above. this will have to be updated to something like this

```
TypeOrmModule.forRootAsync({
      imports: [ConfigModule],
      useFactory: (config: ConfigService) => ({
            host: config.get('DATABASE_HOST'),
            ...
      })
      inject: [ConfigService],
}),
```
</blockquote></details>

We could also make a `database.providers.ts` file like the recipe you sent me does, and do it in there instead. It would make it easier for future teams to connect to multiple databases if for some godforsaken reason they feel the need to do that. 
___
*[`src/inventory/inventory.entity.ts` line 14 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcWRCS9Exautx9DDXp:-OHd62O6DZ7XBI8l5_t7:b-299yka) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/inventory/inventory.entity.ts#L14)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

again, any reasoning for all these defined lengths?
</blockquote></details>

They were chosen during DB design meetings early on, they can be changed whenever. 
___
*[`src/inventory/inventory.entity.ts` line 20 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcWftqB6Rwn1q1oaHP:-OHd69D0CQkxYd6hdMFf:b-6uofyn) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/inventory/inventory.entity.ts#L20)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

not too sure if 50 characters is enough for address line 1
</blockquote></details>

We can increase this. 
___
*[`src/inventory/interfaces/inventory.interface.ts` line 1 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcWtdA0zcCmYc5wcsU:-OHd6Ttn9oPTezMAo4r2:bhy9ost) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/inventory/interfaces/inventory.interface.ts#L1)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

if its not needed, I would just delete this so that there's no confusion in the future
</blockquote></details>

Will do!
___
*[`src/invite/invite.entity.ts` line 5 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcXUt8CM0ULvjtoCDK:-OHd6vXH0V6b0cyPYNoO:bvi86av) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L5)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

im assuming this is how we are going to differentiate from user invites to an org, and org invites to the site. im not too sure if this is how we want to do this.
</blockquote></details>

It's what we discussed during the meetings, and I think its fine. What would the alternatives be?
___
*[`src/invite/invite.entity.ts` line 18 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcY4kgDczlzJZmZ7At:-OHd70oM1tnO9b_Bxqic:b-896fix) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L18)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

maybe inviter?
</blockquote></details>

Done.
___
*[`src/invite/invite.entity.ts` line 25 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcY8M27bI-Sk7ODczg:-OHd70dvENlao4sNc1Mh:b-896fix) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L25)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

maybe inviteeEmail?
</blockquote></details>

Done.
___
*[`src/invite/invite.entity.ts` line 31 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcYC0s1qRFMYXpv_uh:-OHd707Y5XrEQUg03XCx:b-896fix) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L31)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

I think `type` would be fine
</blockquote></details>

Done.
___
*[`src/invite/invite.entity.ts` line 36 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcYGSDB__6_2eyxH32:-OHd6hAD-QYf5ux5Y3r4:bnz14b3) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.entity.ts#L36)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

this is cool
</blockquote></details>

It would be cool if it worked but it currently doesn't. AFAICT it should, but manually checking the DB values shows its just doing CURRENT_TIMESTAMP again.
___
*[`src/invite/invite.module.ts` line 10 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcvrI931HXjBaj10k8:-OHd7MKv0tDmqCAPlW_i:b-e3mw34) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/invite/invite.module.ts#L10)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

why would the invite module need user and organization?
</blockquote></details>

In order to insert an invite into the DB we need valid User and Organization objects for the foreign keys.
___
*[`src/listing/listing.entity.ts` line 36 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcw0ytCVIxx1O8NdWR:-OHd7Yun5w0lLTNdF3N4:b-m4d7so) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/listing/listing.entity.ts#L36)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

there was a default of 0 for inventory item, I think. any reason why this is different?
</blockquote></details>

How would someone create a listing with 0 items? 
___
*[`src/listing/listing.entity.ts` line 39 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcw7RQ1-WDj2ZDaYfv:-OHd7ba9Cwh9h9IKjN3k:b-m77ehv) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/listing/listing.entity.ts#L39)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

I think this is how we want to do this
</blockquote></details>

PostgreSQL has a built in POINT type which we could use instead. It has built in queries to get things like "All other points within 10 miles". 
___
*[`src/organization/organization.entity.ts` line 20 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwUUl4OpIB0QceaNP:-OHd7odxDMRivYtmGeGy:bnv0oom) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L20)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

we should be fine if organizations have the same name. I think we will want to find something else to differentiate them, like tax id
</blockquote></details>

Yeah we discussed this, and it's partially an old comment I forgot to remove. The above part is in reference to having owner be unique, which would mean each person could only have one organization. It feels like something they would want but I'm not sure, so I just left the comment and we can ask later if the team thinks it's worth asking.
___
*[`src/organization/organization.entity.ts` line 26 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwbfJ8j4f3wndViLR:-OHd8XmQDVXshIbQTMrg:b-kff4l5) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L26)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

what does this refer to?
</blockquote></details>

They wanted us to keep track of the last activity for each organization, and to notify them if an org isn't active for X amount of time. (they said 90 days during the meeting, but didn't seem confident with that number). They don't want inactive orgs on the site, so they can remove the ones who aren't using it. 
___
*[`src/organization/organization.entity.ts` line 32 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwgBs-nXW7OOe-c6s:-OHd8ira9c7mqLHHdvNn:b-nxgoxe) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L32)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

how would this look like?
</blockquote></details>

Unsure, it isn't something we discussed in depth yet.
___
*[`src/organization/organization.entity.ts` line 35 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwkuJ05gq_2CrljEZ:-OHd8z-A86qKFqBLOyah:b-bff2wz) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L35)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

how will this be updated?
</blockquote></details>

Unsure, it isn't something we've discussed much yet.
___
*[`src/organization/organization.entity.ts` line 59 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcwpRD41w2vPMgB7sv:-OHd8sQC1lc7_uOKkH1G:b-w6ss5k) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/organization/organization.entity.ts#L59)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

should this be a json?

```
{
    website: "...",
    twitter: "...",
    ....=
}
```
</blockquote></details>

I see no reason why we couldn't make it a JSONB. 
___
*[`src/request/request.entity.ts` line 11 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcx8fM3kP9_yM27WsT:-OHd9B2X12CMvjC2Xou-:b-jp5sz) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/request/request.entity.ts#L11)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

should organization name be on here too? I wonder if we should include organization details here to easily populate those fields to create an entry if the request is approved.
</blockquote></details>

I think that's a good call. It helps the approver know if it should be approved or not because they have more than just the ein to work with. Added.
___
*[`src/user/user.entity.ts` line 17 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcxgQPDCEBn0wCNVFF:-OHdABaJ2fKbeRVjjg6u:b-7bt8xg) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L17)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

how will password and email look if the user is using a different form of authentication?
</blockquote></details>

I believe we discussed that they will be made null, but I could be mistaken. If that's how we want to do it, I can update the columns.
___
*[`src/user/user.entity.ts` line 17 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcyoLu1khFodmQXEWB:-OHdDVD30N9O6SVRSsdF:b-gfvver) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L17)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

I think conversations is missing
</blockquote></details>

It is missing, because I didn't set up the relation between user and conversations yet. I mentioned this in the chat, partially because it might have been ideal to use a ManyToMany relation (which I'm not experienced with yet), and partially because the PR was big enough as is and I didn't want to add any more to it. I have a few commits on a branch off of this and conversation is added in one of those. 
___
*[`src/user/user.entity.ts` line 44 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcxnak0AYh0gY9ZHtv:-OHdAKCP2aGYbEuUCKEt:b-896fix) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L44)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

change this to `type`
</blockquote></details>

Done.
___
*[`src/user/user.entity.ts` line 50 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcxqhI9P9ziPyhU-Ut:-OHd9p3mDiM20EgUZ5s2:blqbx1n) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L50)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

do we need a field to determine if the user is inactive?
</blockquote></details>

They want to know activity statuses on users.
___
*[`src/user/user.entity.ts` line 66 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcyaI405O8ZFl1-ZuY:-OHdDl1j2_9RUuJH4r13:bw661qb) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L66)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

what will be the difference between `referrals` and `sentInvites`?
</blockquote></details>

I'm not honestly sure. We have them both listed separately in the db design, I'll ask Jaime. 
___
*[`src/user/user.entity.ts` line 84 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcyYEo59laBC2XGysy:-OHdAni35dXbVOi0PbQb:bumst8u) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L84)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

why is request an array?
</blockquote></details>

Because we have a relation between two entities. Each request has an approver, and one user can approve many requests. If we wanted to search and get a list of requests someone approved, all we would have to do is `user.approvedRequests`. TypeORM won't grab the list of approved requests by default, we either have to explicitly ask it to or set the relation to be eager. 
___
*[`src/user/user.entity.ts` line 87 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcykHL4EqcvByf3q8k:-OHdBJeLBJW5cRD3v4qq:bepvr6s) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L87)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

is this necessary?
</blockquote></details>

This isn't an actual column in the DB and won't slow down any transactions, it just lets us reverse the relation between user and messages easily. 
___
*[`src/user/user.entity.ts` line 90 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcygO31LIYvy4JdswZ:-OHdBbM4ApkoDVbQdtn-:b-goohwo) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/user.entity.ts#L90)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

does a user have listings, or does an organization?
</blockquote></details>

According to our current schema, a user does. This way, an organization owner can keep track of which people in their organization are the ones making listings. 
___
*[`src/user/dto/create-user.dto.ts` line 1 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcysQsEhegCS8w3Vb8:-OHdBspm3fskjJqfcpKv:b-896fix) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/user/dto/create-user.dto.ts#L1)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

delete it for now
</blockquote></details>

Done.
___
*[`src/conversation/message/message.entity.ts` line 13 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcV0DEBTi9lVeMu_2E:-OHd43pvDd64lt_kJ903:b-3q8alx) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.entity.ts#L13)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

I wonder if we should change this `author`
</blockquote></details>

I agree, I'll change that. 
___
*[`src/conversation/message/message.entity.ts` line 19 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcUeA6-PGg_8F2MoWV:-OHd4H1yDv5ReQZDm-cM:b-s5hy87) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.entity.ts#L19)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

reasoning for this defined length?
</blockquote></details>

It was an arbitrary number chosen when designing the DB. We can set it to whatever we want really. 
___
*[`src/conversation/message/message.entity.ts` line 22 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcUv_k6z0d6ym2_wJh:-OHd4MLxBwtBay7KF1Gv:bwjqhvo) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.entity.ts#L22)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

why is readStatus a date?
</blockquote></details>

I do not remember, but that's how it is in the DB documentation.
___
*[`src/conversation/message/message.module.ts` line 10 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcVBQH3WuHUs-l50KX:-OHd4g2k22Us7pf9buZ4:b-dpzrq2) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.module.ts#L10)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

why would the message module need the user and conversation module?
</blockquote></details>

Because when inserting a message into the db in the service module, we need to insert a valid User object and Conversation object to fill in the foreign key columns. Unless we're sending the objects in the DTO (which is not currently set up) we need to pull them from the db itself. If we go the DTO route we can easily change that in the future, but for the purpose of manually validating everything works correctly I needed them for now. 
___
*[`src/conversation/message/message.service.ts` line 22 at r2](https://reviewable.io/reviews/The-Mobility-Independence-Foundation/backend/6#-OHcVNK68215i-67wYvT:-OHd555jAsgWvUTcyZZ2:bw3nqkm) ([raw file](https://github.com/The-Mobility-Independence-Foundation/backend/blob/aa7df615e213cff28518e8f4a6e8aeeb33e30d26/src/conversation/message/message.service.ts#L22)):*
<details><summary><i>Previously, notcruz (jon) wrote…</i></summary><blockquote>

will this be updated in the future?
</blockquote></details>

Yes, I imagine most everything in these files will be updated in the future. I don't think we would ever need a call to get literally all messages, but we would use one to get all messages in a conversation. 


<!-- Sent from Reviewable.io -->

aas1984 committed:
style: made changes jon requested

Made some variables readonly that weren't previously.

Renamed some columns to be more succinct.

Changed some varchars to have better lengths.

Deleted some unnecessary files and comments.

aas1984 committed:
style: made changes jon requested

Made some variables readonly that weren't previously.

Renamed some columns to be more succinct.

Changed some varchars to have better lengths.

Deleted some unnecessary files and comments.

aas1984 committed:
bugfix: fixed rebase issue in package-lock.json

aas1984 committed:
style: ran npm run format

aas1984 committed:
Merge branch 'type-orm' of https://github.com/The-Mobility-Independence-Foundation/backend into type-orm
